plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java

```diff
@@ -574,29 +592,31 @@ private void onConnectingFailed() {
      * Destroys a machine based on the selected recipe.
      */
     private void disconnect() {
-        if (selectedTarget == null || !selectedTarget.isConnected()) {
-            return;
-        }
-
-        String machineId = null;
-        for (MachineDto machine : machines) {
-            if (machine.getConfig().getName().equals(selectedTarget.getName()) &&
-                    "ssh".equals(machine.getConfig().getType()) &&
-                    machine.getStatus() == MachineStatus.RUNNING) {
-                machineId = machine.getId();
-                break;
+        if (selectedTarget != null && selectedTarget.isConnected()) {
+            for (MachineDto machine : machines) {
+                if (machine.getConfig().getName().equals(selectedTarget.getName()) &&
+                        "ssh".equals(machine.getConfig().getType()) &&
```

Comment:
Let we leave the code as it is. I have calculated 15 string constants in the class and not sure we have to move them to constants.